cache/s3/s3.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"io"
+	"log"
+
+	"github.com/buchgr/bazel-remote/cache"
+	"github.com/minio/minio-go"
+)
+
+type s3Cache struct {
+	mclient  *minio.Client
+	location string
+	bucket   string
+}
+
+// New erturns a new instance of the S3-API based cached
+func New(endpoint string, bucket string, location string,
```

Comment:
This information looks like S3CloudStorageConfig, why not pass in that struct rather than having so many parameters?

Reply:
Yeah makes sense. Done. Also fixed code to pass tests.